src/main/webapp/app/guided-tour/guided-tour.service.ts

```diff
@@ -219,38 +225,76 @@ export class GuidedTourService {
     }
 
     /**
-     * Pause tour for a smooth user interaction
+     * Enable a smooth user interaction
      * @param targetNode an HTMLElement of which DOM changes should be observed
+     * @param userInteraction the user interaction to complete the tour step
      */
-    public pauseTour(targetNode: HTMLElement): void {
+    public enableUserInteraction(targetNode: HTMLElement, userInteraction: UserInteractionEvent): void {
         if (!this.currentTour) {
             return;
         }
         const nextStep = this.currentTour.steps[this.currentTourStepIndex + 1];
-        const observer = new MutationObserver(mutations => {
-            mutations.forEach(() => {
-                if (nextStep) {
-                    document.body.classList.remove('tour-open');
-                    this.guidedTourCurrentStepSubject.next(null);
-                    this.resumeTour(observer, nextStep);
-                }
+
+        if (nextStep && userInteraction === UserInteractionEvent.WAIT_FOR_SELECTOR) {
+            if (nextStep.highlightSelector) {
+                this.waitForElement(nextStep.highlightSelector);
+            } else {
+                this.enableNextStepClick();
+            }
+        } else {
+            const observer = new MutationObserver(mutations => {
```

Comment:
Complicated stuff, good job on finding the `MutationObserver`.

Two ideas:
- We could write a service that uses the MutationObserver and wraps the harder to read callback api in an Observable (e.g. flatMap on the mutations). This way you could do stuff like this:

`observeMutations(node).pipe(filter(m -> m.type !== XY), takeFirst()).subscribe();

- We probably have to pay attention that we always disconnect the MutationObserver in the end to avoid memory leaks (there is no default case here)

Reply:
@thilo-behnke That's a great idea. I can definitely try improve the logic with using Observables. I would do this in another following PR though